pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -598,6 +636,126 @@ func (r *defaultKialiReconciler) reconcileKiali(ctx context.Context, kialiCRName
 	return nil
 }
 
+type PrometheusReconciler interface {
+	reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, configuredMembers []string) error
+}
+
+type defaultPrometheusReconciler struct {
+	Client client.Client
+}
+
+func (r *defaultPrometheusReconciler) reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, members []string) error {
+	reqLogger := common.LogFromContext(ctx)
+	reqLogger.Info("Attempting to get Prometheus ConfigMap", "prometheusNamespace", prometheusNamespace, "prometheusCMName", prometheusCMName)
+
+	cm := &corev1.ConfigMap{}
+
+	err := r.Client.Get(ctx, client.ObjectKey{Name: prometheusCMName, Namespace: prometheusNamespace}, cm)
+	if err != nil {
+		if meta.IsNoMatchError(err) || errors.IsNotFound(err) || errors.IsGone(err) {
+			reqLogger.Info("Prometheus Config Map does not exist.")
+			return nil
+		}
+		return pkgerrors.Wrap(err, "error retrieving Prometheus ConfigMap from mesh")
+	}
+
+	data := make(map[string]interface{})
+
+	err = yaml.Unmarshal([]byte(cm.Data[prometheusConfigurationFilename]), &data)
+
+	if err != nil {
+		return pkgerrors.Wrap(err, "error unmarshaling prometheus.yml")
+	}
+
+	f, ok, err := unstructured.NestedFieldNoCopy(data, prometheusScrapeConfigKeyName)
+
+	var scrapes []interface{}
+
+	if err == nil && ok {
+		scrapes, ok = f.([]interface{})
+
+		if !ok {
+			return fmt.Errorf("error no scrape_configs found from %v", f)
+		}
+	} else {
+		return pkgerrors.Wrap(err, "error getting scrape_configs from ConfigMap")
+	}
+
+	for _, s := range scrapes {
+		scrape, ok := s.(map[string]interface{})
+
+		if !ok {
+			return fmt.Errorf("error converting scrape_config from %v", s)
+		}
+
+		if scrape["job_name"] == "pilot" {
+			continue
+		}
+
+		f, ok, err := unstructured.NestedFieldNoCopy(scrape, prometheusScrapeSDConfigKeyName)
+
+		if err == nil {
```

Comment:
When you handle errors first, then you have less nested code and less indentations.
```suggestion
		if err != nil {
		        return pkgerrors.Wrap(err, fmt.Sprintf("error getting sd from %v", s))
		}
		if ok {
		        ...
```
It would be more readable in my opinion. WDYT?